### PR TITLE
bpo-36957: Add _PyLong_Rshift() and _PyLong_Lshift().

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
They can be used in `isqrt()` and `factorial()` (and in future `perm()` and `comb()`).

https://bugs.python.org/issue36957

### CODE REVIEW

File: Objects/longobject.c
Comment (mdickinson) [inline Objects/longobject.c:4582]: Could we have a short "docstring" comment (one line would be enough) above this function explaining what it does? Of course, it's mostly obvious from the name and context, but I think it's still useful to have that short text description. Same for all the other new functions, if possible, but I think the `_PyLong*` functions intended to be used outside `longobject.c` definitely deserve some kind of docstring.

File: Objects/longobject.c
Comment (mdickinson) [inline Objects/longobject.c]: Nitpick: why the reformatting here, but not in the same line in `long_lshift`?

**Review by mdickinson — APPROVED** [review]
LVGTM! One consistency nitpick, and a request for docstring-like comments on the new `_PyLong_*` functions.

This will probably conflict with #13405, but I think it should be easy to fix the conflicts in #13405 (and of course update #13405 to use the new functions) after this is merged.

**Comment by mdickinson** [comment]
One question: is this worth a news entry? The new functions may be of general interest to other core devs.

File: Objects/longobject.c
Comment (serhiy-storchaka) [reply Objects/longobject.c]: It was not reformatted. The original code in `long_rshift()` was split on two lines, while in `long_lshift()` it was one line. It is GidHub that shows the diff with wrong functions because some code was moved.

**Comment by serhiy-storchaka** [comment]
> One question: is this worth a news entry?

I do not want to encourage using these functions in third-party code. They purpose is the use in the integer-specific functions in the `math` (or `imath`) module and other numbers implementations (`float`). There were no much uses of `PyNumber_Rshift()` and `PyNumber_Lshift()`.

**Comment by mdickinson** [comment]
LGTM!

File: Objects/longobject.c
Comment (mdickinson) [reply Objects/longobject.c]: Makes sense. Thanks.

**Pull request merged**
